File: jasmine/lib/jasmine/locale.ex
defmodule Jasmine.Locale do
  use Ecto.Schema
  import Ecto.Changeset

  schema "locales" do
    field :lang, :string
    field :code, :string
    field :message, :string
    field :version, :integer
    field :updated_at, :utc_datetime_usec
    field :created_at, :utc_datetime_usec
  end

  @doc false
  def changeset(locale, attrs) do
    locale
    |> cast(attrs, [:lang, :code, :message, :version, :updated_at, :created_at])
    |> validate_required([:lang, :code, :message])
  end
end

File: dahlia/src/main/kotlin/com/github/saturn_siv/palm/dahlia/plugins/cms/models/Locale.kt
package com.github.saturn_siv.palm.dahlia.plugins.cms.models

import jakarta.persistence.*
import java.time.Instant

@Table(name = "locales")
@Entity(name = "locale")
class Locale(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    var id: Int,
    @Column(nullable = false)
    var lang: String,
    @Column(nullable = false)
    var code: String,
    @Column(nullable = false)
    var message: String,
    @Column(nullable = false)
    var version: Int,
    @Column(nullable = false)
    var updatedAt: Instant,
    @Column(nullable = false)
    var createdAt: Instant,
)